src/backend/distributed/utils/node_metadata.c

```diff
@@ -475,6 +476,16 @@ ActivateNode(char *nodeName, int nodePort)
 		EnsureNoModificationsHaveBeenDone();
 		ReplicateAllDependenciesToNode(nodeName, nodePort);
 		ReplicateAllReferenceTablesToNode(nodeName, nodePort);
+
+		/*
+		 * Let the maintanince deamon do the hard work of syncing the metadata. We prefer
+		 * this because otherwise node activation mig
+		 */
+		if (ClusterHasDistributedFunction())
```

Comment:
It's worth trying to run the regression tests with metadata syncing always on to check for MX bugs.

e.g. I spotted a crash in `ALTER TABLE collections ATTACH PARTITION collections_4 FOR VALUES IN ( 4 );`

Reply:
We should probably also check the replication model here.

Reply:
When I cherry-pick the commit in #3008 and #3009 and run the regression tests with MX enabled, the crashes are gone. The output difference in the failing tests are just expected.